src/indexes/createIndexes.ts

```diff
@@ -43,20 +47,23 @@ export function createIndexes<T extends IndexableObj>(
   >(isNotUndefined);
 
   const captureObject: Captor<T> = (obj) => {
-    const proxy = new Proxy(Object.assign(obj, {
-      deleteFromIndex() {
-        deleters.forEach(deleteFromIndexFunc => deleteFromIndexFunc(obj))
+    const proxy = new Proxy(
+      Object.assign(obj, {
+        deleteFromIndex() {
+          deleters.forEach((deleteFromIndexFunc) => deleteFromIndexFunc(proxy));
```

Comment:
This is the only line that I changed, pass `proxy` instead of `obj` to `deleteFromIndexFunc`.